### DISCUSSION TITLE
IP归属地为CN，提示问题

### DISCUSSION THREAD

**Discussion by ttyy33** [nextai-translator/nextai-translator#522] [Q&A]
![image](https://user-images.githubusercontent.com/127597621/228406310-da49b347-9cc1-4483-9b8c-22c523e2674f.png)
想问下开发者，是基于什么原理检测的。我是把openai.com的域名和openai translator.exe的进程都设置了代理。
请问这个是基于什么原理，我想去掉它，看着有点烦，或者说能否加一个开关，让用户控制是否显示它。

**Comment by celerysoft** [comment]
请检查一下你请求[https://chat.openai.com/cdn-cgi/trace](https://chat.openai.com/cdn-cgi/trace)时是否使用了代理。如果没有，请添加一条规则，使该请求通过代理进行。

**Reply by ttyy33** [reply]
> 请检查一下你请求https://chat.openai.com/cdn-cgi/trace时是否使用了代理。如果没有，请添加一条规则，使该请求通过代理进行。

我这里只有loc显示CN(改浏览器的归属地？是这个问题嘛，大佬 )我用的clash，我的IP是日本的地址

**Reply by hongyi-zhao** [reply]
[相关信息如下](https://chat.openai.com/cdn-cgi/trace)：

```
fl=618f2
h=chat.openai.com
ip=2a06:a005:2e2:1eab:772e:8c5:9576:3328
ts=1680082262.583
visit_scheme=https
uag=Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/111.0
colo=LAX
sliver=none
http=http/2
loc=US
tls=TLSv1.3
sni=plaintext
warp=off
gateway=off
rbi=off
kex=X25519
```

但是仍然遇到这里的问题：

![image](https://user-images.githubusercontent.com/11155854/228491493-1c35a1ef-efb6-4ec4-9020-d6aad46e0933.png)

**Reply by ttyy33** [reply]
> loc=US

你的这个是怎么改的，大佬

**Reply by hongyi-zhao** [reply]
我是加载代理后使用的。自然显示为代理国家。

**Reply by celerysoft** [reply]
@hongyi-zhao 如果你是用userscript，升级到v0.0.36即可

@ttyy33 需要修改你的clash配置文件，为openai.com添加一条规则，走代理访问

**Reply by ttyy33** [reply]
> @hongyi-zhao 如果你是用userscript，升级到v0.0.36即可
> 
> @ttyy33 需要修改你的clash配置文件，为openai.com添加一条规则，走代理访问

我加了啊，大佬

**Reply by hongyi-zhao** [reply]
> > @hongyi-zhao 如果你是用userscript，升级到v0.0.36即可

你是说的下面这个文件吗？

https://github.com/yetone/openai-translator/releases/download/v0.0.36/openai-translator-0.0.36.user.js

我当时看到了这个配置文件，但是不太清楚具体怎么使用。尝试查找文档，也没有看到特别明显地说明。

所以，我目前的代理方法是基于[proxychains-ng](https://github.com/rofl0r/proxychains-ng) 实现的。

> > @ttyy33 需要修改你的clash配置文件，为openai.com添加一条规则，走代理访问

我没有使用clash， 而是使用的 [haproxy](https://github.com/haproxy/haproxy) 做代理负载均衡和转发控制. 
 
> 我加了啊，大佬 ![image](https://user-images.githubusercontent.com/127597621/228531642-47c74ca5-0ca5-4533-9760-3451adc6adaf.png)

你已经看到，即使我的代理地址显示为 US，仍会出现那个IP地址的警告信息。尽管如此，却并不影响使用。

**Comment by celerysoft** [comment]
@ttyy33 
在
`- PROCESS-NAME,OpenAI Translator.exe,闪电`
下面新增一条配置
`- PROCESS-NAME,msedgewebview2.exe,闪电`
闪电符号我打不出来，见笑

**Reply by hongyi-zhao** [reply]
在Linux系统上，只看到下面的程序：
```shell
werner@X10DAi:~$ which open-ai-translator 
/usr/bin/open-ai-translator
```

**Reply by celerysoft** [reply]
Linux下我不太熟悉，帮不到你，不好意思。不过你可以尝试提个issue试试

**Reply by hongyi-zhao** [reply]
多谢提醒。但是，这个项目的 issue 预定义类别非常奇怪，并没有一般的 issue 可以选择：

![image](https://user-images.githubusercontent.com/11155854/228541852-d4418202-27e6-49fc-9c04-ca5744663d3a.png)

**Reply by ttyy33** [reply]
> @ttyy33 在 `- PROCESS-NAME,OpenAI Translator.exe,闪电` 下面新增一条配置 `- PROCESS-NAME,msedgewebview2.exe,闪电` 闪电符号我打不出来，见笑

我加了，刚测试了以下，还是不行。还是loc=CN。我用的win10 企业版 ltsc 21H2的系统。clash用的tun模式（tun stack=system），麻烦大佬再帮我看下什么问题，麻烦大佬了，谢谢。

**Reply by celerysoft** [reply]
加上配置之后，重启一次软件，如果还是不行，只能建议你先安装低版本试试了

或者说打开clash的Logs页面，筛选chat.openai.com看看走的是什么规则

**Reply by ttyy33** [reply]
> 加上配置之后，重启一次软件，如果还是不行，只能建议你先安装低版本试试了
> 
> 或者说打开clash的Logs页面，筛选chat.openai.com看看走的是什么规则

我把edge浏览器卸载，装成dev版本了，Microsoft Edge WebView2 Runtime也卸载重装了，现在是111.0.1661.54版本。
chat.openai.com 这个日志里一直走的代理。我发现一个有趣的现象，刚才发现的
走ipv6 显示loc=JP显示日本
走ipv4，显示loc=CN显示中国
可以它是一个地址啊
还有我现在open ai translator.exe，现在死活打不开了，可能跟我卸载了edge浏览器和Microsoft Edge WebView2 Runtime有关，求大佬指导以下。

**Reply by celerysoft** [reply]
软件打不开了可以参考这个issue里的解决方法[https://github.com/yetone/openai-translator/discussions/481](https://github.com/yetone/openai-translator/discussions/481)

照你说的这个情况，应该是openai那边把你ipv4的ip错误地识别成中国的ip了，只能换个代理ip，或者看看开发者后续会不会去除这个限制。

在此之前，你先用用老版本替代一下，我看了下，v0.0.34及更早的版本都没有添加这个ip限制，你可以试试

顺便，你的ip暴露了，我看了你第一个回复，你把ip做了模糊处理，所以，也处理下这个回复吧

**Reply by ttyy33** [reply]
> 软件打不开了可以参考这个issue里的解决方法#481
> 
> 照你说的这个情况，应该是openai那边把你ipv4的ip错误地识别成中国的ip了，只能换个代理ip，或者看看开发者后续会不会去除这个限制。
> 
> 在此之前，你先用用老版本替代一下，我看了下，v0.0.34及更早的版本都没有添加这个ip限制，你可以试试
> 
> 顺便，你的ip暴露了，我看了你第一个回复，你把ip做了模糊处理，所以，也处理下这个回复吧

感谢大佬，按照你发的链接，能正常打开,  现在不会提示我了，感谢大佬的一步帮助和指导
